Src/Amr/AMReX_extrapolater_2D_K.H

```diff
@@ -0,0 +1,234 @@
+                             amrex::Array4<const int>   const& mask,
+                             amrex::Array4<amrex::Real> const& data) noexcept
+{
+   using namespace amrex::literals;
+
+   constexpr int finecell = 1;
+   constexpr int crsecell = 0;
+
+   const auto lo = amrex::lbound(bx);
+   const auto hi = amrex::ubound(bx);
+
+   int k = lo.z;
+
+   for (int n = 0; n < nComp; n++) {
+
+      // set all crse cells to zero first
+      for (int j = lo.y-1; j <= hi.y+1; ++j) {
+         for (int i = lo.x-1; i <= hi.x+1; ++i) {
+            if (mask(i,j,k) == crsecell) data(i,j,k,n) = 0.0_rt;
+         }
+      }
+
+      // Corners
+      // xlo, ylo
+      {
+         int i = lo.x-1;
+         int j = lo.y-1;
+         if ( mask(i,j,k) == crsecell ) {
+            if ( ( mask(i+1,j,k) == finecell ) ||
+                 ( mask(i,j+1,k) == finecell ) ) {
+               data(i,j,k,n) = ( mask(i+1,j,k) * data(i+1,j,k,n) + mask(i,j+1,k) * data(i,j+1,k,n) )
+                               / ( mask(i+1,j,k) + mask(i,j+1,k) );
+            } else {
+               data(i,j,k,n) = data(i+1,j+1,k,n);
+            }
+         }
+      }
+      // xlo, yhi
+      {
+         int i = lo.x-1;
+         int j = hi.y+1;
+         if ( mask(i,j,k) == crsecell ) {
+            if ( ( mask(i+1,j,k) == finecell ) ||
+                 ( mask(i,j-1,k) == finecell ) ) {
+               data(i,j,k,n) = ( mask(i+1,j,k) * data(i+1,j,k,n) + mask(i,j-1,k) * data(i,j-1,k,n) )
+                               / ( mask(i+1,j,k) + mask(i,j-1,k) );
+            } else {
+               data(i,j,k,n) = data(i+1,j-1,k,n);
+            }
+         }
+      }
+      // xhi, ylo
+      {
+         int i = hi.x+1;
+         int j = lo.y-1;
+         if ( mask(i,j,k) == crsecell ) {
+            if ( ( mask(i-1,j,k) == finecell ) ||
+                 ( mask(i,j+1,k) == finecell ) ) {
+               data(i,j,k,n) = ( mask(i-1,j,k) * data(i-1,j,k,n) + mask(i,j+1,k) * data(i,j+1,k,n) )
+                               / ( mask(i-1,j,k) + mask(i,j+1,k) );
+            } else {
+               data(i,j,k,n) = data(i-1,j+1,k,n);
+            }
+         }
+      }
+      // xhi, yhi
+      {
+         int i = hi.x+1;
+         int j = hi.y+1;
+         if ( mask(i,j,k) == crsecell ) {
+            if ( ( mask(i-1,j,k) == finecell ) ||
+                 ( mask(i,j-1,k) == finecell ) ) {
+               data(i,j,k,n) = ( mask(i-1,j,k) * data(i-1,j,k,n) + mask(i,j-1,k) * data(i,j-1,k,n) )
+                               / ( mask(i-1,j,k) + mask(i,j-1,k) );
+            } else {
+               data(i,j,k,n) = data(i-1,j-1,k,n);
+            }
+         }
+      }
+      // Edges
+      // xlo, y-valid
+      {
+         int i = lo.x-1;
+         for (int j = lo.y; j <= hi.y; ++j) {
+            if ( mask(i,j,k) == crsecell ) {
+               data(i,j,k,n) = ( mask(i,j-1,k) * data(i,j-1,k,n) + mask(i+1,j,k) + mask(i,j+1,k) * data(i,j+1,k,n) )
```

Comment:
This should be `data(i,j,k,n) = ( mask(i,j-1,k) * data(i,j-1,k,n) + data(i+1,j,k) + mask(i,j+1,k) * data(i,j+1,k,n) )`.

Reply:
Note that the typo is in GPU version too.

Reply:
My eyes clearly had trouble focusing there ... too much i,j,k, +1, -1 ... I'll fix those and go over it again.